cognite/client/_api/transformation_notifications.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Optional, Union
+
+from cognite.client import utils
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes import (
+    OidcCredentials,
+    Transformation,
+    TransformationBlockedInfo,
+    TransformationDestination,
+    TransformationList,
+    TransformationNotification,
+    TransformationNotificationList,
+)
+from cognite.client.data_classes.transformation_notifications import TransformationNotificationFilter
+
+
+class TransformationNotificationsAPI(APIClient):
+    _RESOURCE_PATH = "/transformations/notifications"
+    _LIST_CLASS = TransformationNotificationList
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def create(
+        self, notification: Union[TransformationNotification, List[TransformationNotification]]
+    ) -> Union[TransformationNotification, TransformationNotificationList]:
+        """`Subscribe for notifications on the transformation errors. <https://docs.cognite.com/api/playground/#operation/subscribeTransformationNotifications>`_
+
+        Args:
+            notification (Union[TransformationNotification, List[TransformationNotification]]): Notification or list of notifications to create.
+
+        Returns:
+            Created notification(s)
+
+        Examples:
+
+            Create new notifications:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> from cognite.experimental.data_classes import TransformationNotification
+                >>> c = CogniteClient()
+                >>> notifications = [TransformationNotification(transformation_id = 1, destination="my@email.com"), TransformationNotification(transformation_external_id="transformation2", destination="other@email.com"))]
+                >>> res = c.transformations.notifications.create(notifications)
+        """
+        utils._auxiliary.assert_type(notification, "notification", [TransformationNotification, list])
+        return self._create_multiple(notification)
+
+    def list(
+        self,
+        transformation_id: Optional[int] = None,
+        transformation_external_id: str = None,
```

Comment:
any particular reason for using internal ids here (and other places too)? we had a discussion last week to expose external ids only. some other core services like labels, relationships etc only expose external id for example.

Reply:
Because transformations are also referred to with ids in the UI and the user is aware that transformations both have id and external IDs. I think it is good to allow them to work with both because of this.

Reply:
We could change that though, and run some migration to make sure all Transformations have an externalId

Reply:
What would we put into the externalId then? name is not unique, this would require us to decide on behalf of user and we exposed the ID in UI already. people see it and use it when reporting problems. So I don't think we should do this personally.